tests/ui_tools/test_popups.py

```diff
@@ -179,6 +180,44 @@ def test_keypress_navigation(
 
         self.super_keypress.assert_called_once_with(size, navigation_key)
 
+    @pytest.mark.parametrize(
+        [
+            "contents",
+            "column_widths",
+            "expected_instance",
+        ],
+        [
+            (
+                [
+                    (
+                        "Viewing Actions",
+                        [
+                            ("Open in web browser", "[]"),
+                            ("Full rendered message", "[]"),
+                            ("Full raw message", "[]"),
+                        ],
+                    )
+                ],
+                [21, 24],
+                [Text, AttrWrap, AttrWrap, AttrWrap],
+            ),
+            ([("", "")], [21, 24], []),
+        ],
+        ids=[
+            "heading_widget_with_row",
+            "widget_with_str",
+        ],
+    )
+    def test_make_table_with_categories(
```

Comment:
Having this test is great, but it does focus on the urwid form of the output, so it may be worth adding a suffix to the test name to emphasize that, eg. `__instances` based on your naming elsewhere (which can be improved).

Reply:
Adding an `__instances` suffix. Although I wonder if this implies there's a test that focuses on `make_table_with_categories` as a whole and not on some specific thing like instances.

Reply:
I agree that it could suggest another test exists, but also means it won't conflict with another later ;) If there are multiple tests for a function, my preferred form now is to separate the function and the intention behind the test.

I suggested `__instances` since that's what you were checking in the test, and used in the expected part of the test case name, as per another comment - so let's sync them up if you think the updated expected parameter name is better :)

Reply:
I'e used `types` everywhere instead of instance because it seems more intuitive than `instances`. So let's change it to `__types` in the function name also?

Reply:
Yeah, that'd be good :+1: